src/app/clusters/actions-server/actions-server.cpp

```diff
@@ -0,0 +1,355 @@
+        return;
+    case Actions::Commands::StartAction::Id:
+        HandleCommand<Commands::StartAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleStartAction(ctx, commandData); });
+        return;
+    case Actions::Commands::StartActionWithDuration::Id:
+        HandleCommand<Commands::StartActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleStartActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::StopAction::Id:
+        HandleCommand<Commands::StopAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleStopAction(ctx, commandData); });
+        return;
+    case Actions::Commands::PauseAction::Id:
+        HandleCommand<Commands::PauseAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandlePauseAction(ctx, commandData); });
+        return;
+    case Actions::Commands::PauseActionWithDuration::Id:
+        HandleCommand<Commands::PauseActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandlePauseActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::ResumeAction::Id:
+        HandleCommand<Commands::ResumeAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleResumeAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableAction::Id:
+        HandleCommand<Commands::EnableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleEnableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableActionWithDuration::Id:
+        HandleCommand<Commands::EnableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleEnableActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableAction::Id:
+        HandleCommand<Commands::DisableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleDisableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableActionWithDuration::Id:
+        HandleCommand<Commands::DisableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleDisableActionWithDuration(ctx, commandData); });
+        return;
+    }
+}
+void MatterActionsPluginServerInitCallback()
+{
+    AttributeAccessInterfaceRegistry::Instance().Register(Instance::GetInstance());
+    CommandHandlerInterfaceRegistry::Instance().RegisterCommandHandler(Instance::GetInstance());
```

Comment:
Yeah, so either we need to have a singleton instance that then needs to be able to handle multiple endpoints, or we need to have instance-per-endpoint.

Reply:
Updated code to have multiple endpoints.